src/Core/CFamily/CFamilyShared.cs

```diff
@@ -31,5 +31,28 @@ public static class CFamilyShared
 
         public static readonly StringComparer RuleKeyComparer = SonarRuleRepoKeys.RepoKeyComparer;
         public static readonly StringComparison RuleKeyComparison = StringComparison.Ordinal;
+
+        /// <summary>
+        /// Attempts to detect whether the file is C or C++ based on the file extension.
+        /// Returns null if the extension is not recognised.
+        /// </summary>
+         public static string FindLanguageFromExtension(string analyzedFilePath)
```

Comment:
I extracted this code from `CFamilyHelper.Capture` class. The only change is to use an `OrdinalIgnoreCase` string comparer, rather than forcing every string to lower case.